s3fs/core.py

```diff
@@ -272,7 +272,7 @@ def get_delegated_s3pars(self, exp=3600):
                 'token': cred['SessionToken'], 'anon': False}
 
     def _open(self, path, mode='rb', block_size=None, acl='', version_id=None,
-              fill_cache=None, cache_type=None, autocommit=True, **kwargs):
+              fill_cache=None, cache_type=None, autocommit=True, req_kw=None, **kwargs):
```

Comment:
Can we call this `request_options`? I had trouble understanding `req_kw` at a glance.

I'd also prefer that this is added to fsspec as an official part of the API.

Reply:
It specifically stands for "requestor-pays" because, although it is a general dict of kwargs, that is the only option it is used for (or maybe the only one possible). Is it different from the other kwargs because it is only needed when getting actual data, not for file listings, metadata, etc.

Having a set of arbitrary things things in the file system layer that get used upon `open` is not a bad idea, but will be tricky to arrange.

Reply:
Just to throw an idea out there - what about having the public API of S3File match that of S3FileSystem, and just have a requester_pays argument on the constructor, keeping req_kw internal?